src/charm.py

```diff
@@ -343,7 +294,7 @@ def _on_tracing_endpoint_removed(self, event) -> None:
             # clear anything up. We could defer, but again, we're being torn down and the unit db
             # will
             return
-        self._clear_tracing_config()
+        self.traefik.clear_tracing_config()
```

Comment:
Not sure if we should keep doing delta-charming here.

Reply:
this PR is not about making traefik fully holistic. 
Happy to consider in a follow-up